Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -144,112 +144,117 @@ public class XPCServer {
             connections.compactMap{ $0.connection }.forEach{ xpc_connection_set_target_queue($0, newValue) }
         }
     }
+    
     // MARK: Route registration
         
-    private func checkUniqueRoute(route: XPCRoute) throws {
-        if self.routes.keys.contains(route) {
-            throw XPCError.routeAlreadyRegistered(route.pathComponents)
+    /// Internal function that actually registers the route and enforces that a route is only ever registered once.
+    ///
+    /// All of the public functions exist to satisfy type constraints.
+    private func registerRoute(_ route: XPCRoute, handler: XPCHandler) {
+        if let _ = self.routes.updateValue(handler, forKey: route) {
```

Comment:
I think this reads more clearly:

```suggestion
    if self.routes.updateValue(handler, forKey: route) != nil {
```